plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -152,48 +158,60 @@ type InliningConstraints tyname name uni fun =
     , PLC.ToBuiltinMeaning uni fun
     )
 
-
+-- | Useful info for inlining.
 data InlineInfo name fun a = InlineInfo
     { _iiStrictnessMap :: Deps.StrictnessMap
+    -- ^ Is it strict? Only needed for PIR, not UPLC
     , _iiUsages        :: Usages.Usages
+    -- ^ how many times is it used?
     , _iiHints         :: InlineHints name a
+    -- ^ have we explicitly been told to inline.
     , _iiBuiltinVer    :: PLC.BuiltinVersion fun
+    -- ^ the builtin version.
     }
 makeLenses ''InlineInfo
 
--- Using a concrete monad makes a very large difference to the performance of this module (determined from profiling)
-type InlineM tyname name uni fun a = ReaderT (InlineInfo name fun a) (StateT (Subst tyname name uni fun a) Quote)
+-- | The monad the inliner runs in.
+-- Using a concrete monad makes a very large difference to the performance of this module
+-- (determined from profiling)
+type InlineM tyname name uni fun a =
+    ReaderT (InlineInfo name fun a) (StateT (Subst tyname name uni fun a) Quote)
 
+-- | Look up the unprocessed variable from the substitution.
 lookupTerm
     :: (HasUnique name TermUnique)
-    => name
-    -> Subst tyname name uni fun a
+    => name -- ^ The name of the variable.
+    -> Subst tyname name uni fun a -- ^ The substitution.
     -> Maybe (InlineTerm tyname name uni fun a)
 lookupTerm n subst = lookupName n $ subst ^. termEnv . unTermEnv
 
+-- | Insert the unprocessed variable to the substitution.
```

Comment:
```suggestion
-- | Insert the unprocessed variable into the substitution.
```

Reply:
And TBH I would say "add a mapping for a type variable to the substitution". You can't put a variable _alone_ in there!